core/src/test/java/io/questdb/log/LogRecordSinkTest.java

```diff
@@ -122,4 +251,163 @@ public void testSimpleMessage3() throws Exception {
+            } finally {
+                Unsafe.free(msgPtr, utf8ByteLen, MemoryTag.NATIVE_DEFAULT);
+            }
+        });
+    }
+
+    @Test
+    public void testUtf8LineTrimmingFromCharSeq() throws Exception {
+        runTestUtf8LineTrimmingImpl((msg, utf8ByteLen, sinkMaxLen, expectedLen) -> {
+            final Utf8 utf8 = new Utf8(msg);
+            final long msgPtr = Unsafe.malloc(utf8ByteLen, MemoryTag.NATIVE_DEFAULT);
+            try {
+                LogRecordSink logRecord = new LogRecordSink(msgPtr, sinkMaxLen);
+                logRecord.put(utf8);
+                Assert.assertEquals(expectedLen, logRecord.length());
+            } finally {
+                Unsafe.free(msgPtr, utf8ByteLen, MemoryTag.NATIVE_DEFAULT);
+                Misc.free(utf8);
+            }
+        });
+    }
+
+    @Test
+    public void testUtf8LineTrimmingFromCharSeqOverload() throws Exception {
+        runTestUtf8LineTrimmingImpl((msg, utf8ByteLen, sinkMaxLen, expectedLen) -> {
+            final String prefix = "test prefix";
+            final String msg2 = prefix + msg;
+            final Utf8 utf8 = new Utf8(msg2);
+            final long msgPtr = Unsafe.malloc(utf8ByteLen, MemoryTag.NATIVE_DEFAULT);
+            try {
+                LogRecordSink logRecord = new LogRecordSink(msgPtr, sinkMaxLen);
+                logRecord.put(utf8, prefix.length(), utf8.length());
+                Assert.assertEquals(expectedLen, logRecord.length());
+            } finally {
+                Unsafe.free(msgPtr, utf8ByteLen, MemoryTag.NATIVE_DEFAULT);
+                Misc.free(utf8);
+            }
+        });
+    }
+
+    @Test
+    public void testUtf8LineTrimmingFromUtf8Buf() throws Exception {
+        runTestUtf8LineTrimmingImpl((msg, utf8ByteLen, sinkMaxLen, expectedLen) -> {
+            final Utf8 utf8 = new Utf8(msg);
+            final long msgPtr = Unsafe.malloc(utf8ByteLen, MemoryTag.NATIVE_DEFAULT);
+            try {
+                LogRecordSink logRecord = new LogRecordSink(msgPtr, sinkMaxLen);
+                final long lo = utf8.address();
+                final long hi = lo + utf8.length();
+                logRecord.putUtf8(lo, hi);
+                Assert.assertEquals(expectedLen, logRecord.length());
+            } finally {
+                Unsafe.free(msgPtr, utf8ByteLen, MemoryTag.NATIVE_DEFAULT);
+                Misc.free(utf8);
+            }
+        });
+    }
+
+    @FunctionalInterface
+    interface ScenarioConsumer {
+        void accept(String msg, int utf8ByteLen, long sinkMaxLen, long expectedLen) throws Exception;
+    }
+
+    /**
+     * A class that holds UTF-8 in native memory.
+     * <p>
+     * Solely intended for testing purposes.
+     * <p>
+     * It conforms to the expections of LogRecordSink
+     * and its length() / charAt() etc methods operate/return
+     * bytes even when the signature says chars.
+     */
+    static final class Utf8 implements LPSZ, CharSequence, Closeable {
```

Comment:
i don't believe we have anything that behaves like this class, pls correct me if i'm wrong